### DISCUSSION TITLE
dissable SSR globally

### DISCUSSION THREAD

**Discussion by idhard** [vercel/next.js#12202] [Help]
Hey , is it possible to disable ssr globally ? i have a big app running client side only libraries using create-react-app and would be lovely to test nextjs static only mode to see how the app behaves 
thanks!

**Comment by phoique** [comment]
then why did you use nextjs.

**Comment by rafaelalmeidatk** [comment]
I am not exactly sure if this is what you want, but if you remove `getInitialProps` from your pages (and `_app`, if it is there) they will automatically be static: https://nextjs.org/docs/advanced-features/automatic-static-optimization

Or you can use `next export` that will generate static files like create-react-app: https://nextjs.org/docs/advanced-features/static-html-export

**Reply by idhard** [reply]
dummy example 

```javascript
import Head from 'next/head'

export default function Home() {
  console.log(window.location);
  return (
    <div className="container">
    </div>
  )
}
```

with SSR  nextjs is complaining about window is not defined , i know you could run side effects hooks to have it work or wrap your components with [no-ssr](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr), but as i said, we have a big app with some libraries that depend of having window available and if we could just have static sites would be a direct replace for create-react-app

**Reply by timneutkens** [reply]
Reading `window.location` inside of the rendering cycle of React is a bad practice as it affects performance, you should use `useEffect` and read it there instead.

**Reply by idhard** [reply]
indeed , is a bad practice , it's just an example , there are many cases you would like to have window object in your client side context independently of react render cycles, ie: an external library which depend on it. 

so i understand is not possible to globally disable SSR , right ?

thanks , and great job with the framework !

**Reply by timneutkens** [reply]
> so i understand is not possible to globally disable SSR , right ?

It's not as it generally leads to worse UX.

E.g. with automatic static optimization you can render skeleton loading states instead of showing a white screen to users.
